src/material/datepicker/calendar-body.ts

```diff
@@ -21,11 +21,13 @@ import {
 } from '@angular/core';
 import {take} from 'rxjs/operators';
 
-/**
- * Extra CSS classes that can be associated with a calendar cell.
- */
+/** Extra CSS classes that can be associated with a calendar cell. */
 export type MatCalendarCellCssClasses = string | string[] | Set<string> | {[key: string]: any};
 
+/** Function that can generate the extra classes that should be added to a calendar cell. */
+export type MatCalendarCellClassFunction<D> =
+    (date: D, view: 'month' | 'year' | 'multi-year') => MatCalendarCellCssClasses;
```

Comment:
Do we have a type alias for `'month' | 'year' | 'multi-year'` somewhere? I thought we did, but I could be crazy

Reply:
I think we've just been doing it inline.